spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1750,6 +1766,59 @@ public Boolean isMandatoryFor(final Message message) {
 			throw convertRabbitAccessException(ex);
 		}
 		finally {
+			if (!invokeScope) {
+				if (resourceHolder != null) {
+					ConnectionFactoryUtils.releaseResources(resourceHolder);
+				}
+				else {
+					RabbitUtils.closeChannel(channel);
+					RabbitUtils.closeConnection(connection);
+				}
+			}
+		}
+	}
+
+	@Override
+	public <T> T invoke(TemplateCallback<T> action) {
+		final Channel currentChannel = this.dedicatedChannels.get();
+		Assert.state(currentChannel == null, () -> "Nested invoke() calls are not supported; channel '" + currentChannel
+				+ "' is already associated with this thread");
+		this.activeTemplateCallbacks.incrementAndGet();
+		RabbitResourceHolder resourceHolder = null;
+		Connection connection = null; // NOSONAR (close)
+		Channel channel;
+		if (isChannelTransacted()) {
+			resourceHolder = ConnectionFactoryUtils.getTransactionalResourceHolder(getConnectionFactory(), true);
```

Comment:
Yes, that crossed my mind - I considered throwing an exception (or not store in the TL) if they attempted to use this technique while in a global transaction but it seemed to me there is no real issue here (except we'd just document it's unnecessary to use `invoke` if there is already a global channel.